### DISCUSSION TITLE
Add @withoutGlobalScopes directive

### DISCUSSION THREAD

**Discussion by saeed-rostami** [nuwave/lighthouse#2574] [Ideas]
Hi
I create @withoutGlobalScopes directive like @scopes that takes an array of scope names ...@withoutGlobalScopes(names: ["first" , "second"])...
i test it and it works.
I would appreciate it if I pushed it to the repository.

**Comment by spawnia** [comment]
There is no `@scopes` directive, only `@scope` - which works on arguments. How would `@withoutGlobalScopes` work? If you want feedback before creating a pull request, please provide a full example.

**Comment by saeed-rostami** [comment]
Yes I mean @scope, not @scopes, my mistake.
@withoutGlobalScopes works on arguments that take an array of global scopes names that are defined in the eloquent model...
for instance, I have an Episode model and I define a global scope name published which only gets published records...


```
   episodes
    (
        allEpisodes: Boolean @withoutGlobalScopes(names: ["published" , "secondScope"])
)    
    : [Episode!]! @all
```
**above allEpisodes argument takes all records (also unpublished)**


[WithoutGlobalScopesDirective.zip](https://github.com/user-attachments/files/15958190/WithoutGlobalScopesDirective.zip)
The WithoutGlobalScopesDirective class is attached, you can check it.
thanks for your time

**Reply by spawnia** [reply]
Looks good. Feel free to send in a pull request. Make sure to add tests.
